src/Raven.Client/Documents/Indexes/IndexDefinitionHelper.cs

```diff
@@ -226,22 +226,42 @@ internal static string GetQuerySource(DocumentConventions conventions, Type type
             return $"{source}[@\"{collectionName.Replace("\"", "\"\"")}\"]";
         }
 
-        internal static IndexType DetectStaticIndexType(string map, string reduce)
+        internal static IndexSourceType DetectStaticIndexSourceType(string map)
         {
             if (string.IsNullOrWhiteSpace(map))
                 throw new ArgumentException("Value cannot be null or whitespace.", nameof(map));
 
-            while (true)
-            {
-                // strip whitespace, comments, etc
-                var m = CommentsStripper.Match(map);
-                if (m.Success == false || m.Index != 0)
-                    break;
+            map = StripComments(map);
+            map = UnifyWhiteSpace(map);
 
-                map = map.Substring(m.Length);
+            // detect first supported syntax: timeseries.Companies.HeartRate.Where
+            if (map.StartsWith("timeSeries"))
+            {
+                return IndexSourceType.TimeSeries;
+            }
+            
+            if (map.StartsWith("from"))
+            {
+                // detect `from ts in timeseries` or `from ts in timeseries.Users.HeartRate`
+                
+                var tokens = map.Split(new [] {' '}, 4, StringSplitOptions.RemoveEmptyEntries);
+                if (tokens.Length >= 4 && tokens[2] == "in" && tokens[3].StartsWith("timeSeries"))
```

Comment:
Please add `StringComparison.OrdinalIgnoreCase` as second parameter - @ppekrol will make it case insensitive

Reply:
done